google/cloud/storage/oauth2/service_account_credentials_test.cc

```diff
@@ -352,6 +359,102 @@ TEST_F(ServiceAccountCredentialsTest, ParseMissingFieldFails) {
   }
 }
 
+/// @test Verify that refreshing a credential updates the timestamps.
+TEST_F(ServiceAccountCredentialsTest, RefreshingUpdatesTimestamps) {
+  auto info = ParseServiceAccountCredentials(kJsonKeyfileContents, "test");
+  ASSERT_STATUS_OK(info);
+
+  auto make_request_assertion = [&info](long timestamp) {
+    return [timestamp, &info](std::string const& p) {
+      std::string const prefix =
+          std::string("grant_type=") + kGrantParamEscaped;
+      EXPECT_THAT(p, HasSubstr(prefix));
```

Comment:
It looks like below we're depending on this to "begin with prefix" rather than "has substr prefix".

Reply:
Fixed. I tried to use `ASSERT_THAT()`, but that makes the lambda return `void` and we want a different return type.